src/plugins/collision/collider/backend.rs

```diff
@@ -131,6 +131,74 @@ impl<C: ScalableCollider> Plugin for ColliderBackendPlugin<C> {
                 );
             });
 
+        // When the `Sensor` component is added to a collider,
+        // remove the collider's contribution on the rigid body's mass properties.
+        app.observe(
+            |trigger: Trigger<OnAdd, Sensor>,
+             query: Query<(
+                &ColliderParent,
+                &ColliderMassProperties,
+                &PreviousColliderTransform,
+            )>,
+             mut body_query: Query<MassPropertiesQuery>| {
+                if let Ok((
+                    collider_parent,
+                    collider_mass_properties,
+                    previous_collider_transform,
+                )) = query.get(trigger.entity())
+                {
+                    if let Ok(mut mass_properties) = body_query.get_mut(collider_parent.0) {
+                        // Subtract previous collider mass props from the body's own mass props.
+                        mass_properties -= ColliderMassProperties {
```

Comment:
Am I understanding the following right?
- When a `Collider` is added, its `ColliderParent` gets mass properties added
- When a `Sensor` is added, its `ColliderParent` gets mass properties removed

Ideally, these two then cancel each other out.
Assuming I got the above correct:
Is the ordering relevant? Can the sensor part be executed first, resulting in a temporary negative collider mass property?

Reply:
If you added `Sensor` first, the entity probably wouldn't have `ColliderParent` or `ColliderMassProperties`, since those are only added automatically for colliders.

If the collider and `Sensor` were added at the same time, with the current setup, the collider wouldn't have mass properties computed for it yet because the observer should run before `init_colliders`. So I believe it shouldn't be an issue currently.

However, if we later changed `init_colliders` to use an observer or lifecycle hook as well, the order would be more ambiguous, which could cause issues. A potential solution could be to initialize sensors with zero mass properties (we already don't run `update_collider_mass_properties` for them anyway) and only subtract from the rigid body's mass properties in the observer if the collider mass properties are non-zero.